lib/Default/AbstractSmrPlayer.class.php

```diff
@@ -2239,12 +2261,23 @@ public function &killPlayerByPlayer(AbstractSmrPlayer $killer) {
 		$return['DeadExp'] = max(0, IFloor($this->getExperience() * $expLossPercentage));
 		$this->decreaseExperience($return['DeadExp']);
 
-		// Killer gains 50% of the lost exp
-		$return['KillerExp'] = max(0, ICeil(0.5 * $return['DeadExp']));
-		$killer->increaseExperience($return['KillerExp']);
+		// newbie kills do not award XP, credits, or military pay
+		if ($this->isNoncombatNewbie() && !$killer->hasNewbieStatus()) {
+			$return['KillerExp'] = $return['KillerCredits'] = 0;
+		} else {
+			// Killer gains 50% of the lost exp
+			$return['KillerExp'] = max(0, ICeil(0.5 * $return['DeadExp']));
+			$killer->increaseExperience($return['KillerExp']);
 
-		$return['KillerCredits'] = $this->getCredits();
-		$killer->increaseCredits($return['KillerCredits']);
+			// Killer gains credits carried
+			$return['KillerCredits'] = $this->getCredits();
+			$killer->increaseCredits($return['KillerCredits']);
+			
+			// War setting gives them military pay
+			if ($relation <= RELATIONS_WAR) {
```

Comment:
`$relation` is not defined until below (see Scrutinizer issue). You'll need to move that definition up here.